keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -491,6 +487,34 @@ TypedValue](https://github.com/kubernetes/kubernetes/blob/9ff3b7e744b34c099c1405
 object has any unknown or duplicate fields.
 
 
+#### Kubectl Flag
+
+For beta, we will modify the kubectl `--validate` flag from being a bool flag to
+a string flag that accepts the following values:
+
+* `server`: This sends a request with the fieldValidation param set to `Strict`
+for strict server-side validation.
+* `client`: This performs client-side validation in kubectl that breaks if it
+encounters any validation errors before sending a request to the server. If no
+client-side validation errors are found, the request sent to the server has
+fieldValidation param set to `Ignore`. This will be the default behavior in beta
+as it is identical to the current validation behavior pre-1.24
+* `ignore`: This performs no client-side validation or server-side validation,
+sending a request to the server with fieldValidation param set to `Ignore`.
+* `warn`: This sends a request to the server with the fieldValidation param set
+  to `Warn`. It performs server-side validation, but validation errors are
+  exposed as warnings in the result header rather than failing the request. This
+  will be the default once server-side validation goes GA.
+* `strict`: same behavior as `--validate=server`.
+* `true`: This is the same as `--validate=client`, but throws a warning along
```

Comment:
I expected `--validate=true` to default to strict server-side validation against servers that advertise support for it in their openapi doc

Reply:
I wasn't sure if changing the behavior of the "true" value to now mean server-side was less disruptive to users than keeping it as is (client-side) but warning that "true/false" semantics are being replaced with "strict/warn/ignore" for server-side.

I don't feel too strongly either way, so I've modified it as you've suggested to make `--validate=true` mean strict server-side validation.

See my related comment below on deprecating "true/false".